src/frontend/app/features/service-catalog/services.service.ts

```diff
@@ -89,36 +126,83 @@ export class ServicesService {
     );
   }
 
-  getVisiblePlans = () => {
+  getServiceBrokerById = (guid: string): Observable<APIResource<IServiceBroker>> => this.serviceBrokers$
+    .pipe(
+      filter(p => !!p),
+      map(brokers => brokers.filter(b => b.metadata.guid === guid)),
+      filter(s => s && s.length === 1),
+      map(s => s[0]),
+      first()
+    )
+
+  getVisibleServicePlans = () => {
     return this.servicePlans$.pipe(
       filter(p => !!p && p.length > 0),
       map(o => o.filter(s => s.entity.bindable)),
-      combineLatest(this.servicePlanVisibilities$),
-      map(([svcPlans, svcPlanVis]) => this.fetchVisiblePlans(svcPlans, svcPlanVis)),
+      combineLatest(this.servicePlanVisibilities$, this.serviceBrokers$, this.service$),
+      map(([svcPlans, svcPlanVis, svcBrokers, svc]) => this.fetchVisiblePlans(svcPlans, svcPlanVis, svcBrokers, svc)),
+
     );
   }
 
   fetchVisiblePlans =
-    (svcPlans: APIResource<IServicePlan>[], svcPlanVis: APIResource<IServicePlanVisibility>[]): APIResource<IServicePlan>[] => {
+    (svcPlans: APIResource<IServicePlan>[],
+      svcPlanVis: APIResource<IServicePlanVisibility>[],
+      svcBrokers: APIResource<IServiceBroker>[],
+      svc: APIResource<IService>): APIResource<IServicePlan>[] => {
       const visiblePlans: APIResource<IServicePlan>[] = [];
       svcPlans.forEach(p => {
         if (p.entity.public) {
           visiblePlans.push(p);
         } else if (svcPlanVis.filter(svcVis => svcVis.entity.service_plan_guid === p.metadata.guid).length > 0) {
+          // plan is visibilities
+          visiblePlans.push(p);
+        } else if (svcBrokers.filter(s => s.metadata.guid === svc.entity.service_broker_guid)[0].entity.space_guid) {
+          // Plan is space-scoped
           visiblePlans.push(p);
         }
       });
       return visiblePlans;
     }
 
-  getServicePlanByGuid = (servicePlanGuid: string) => {
-    return this.servicePlans$.pipe(
-      filter(p => !!p),
-      map(servicePlans => servicePlans.filter(o => o.metadata.guid === servicePlanGuid)),
-      map(o => o[0])
+  getServicePlanAccessibility = (servicePlan: APIResource<IServicePlan>): Observable<ServicePlanAccessibility> => {
+    if (servicePlan.entity.public) {
+      return Observable.of({
+        isPublic: true,
+        guid: servicePlan.metadata.guid
+      });
+    }
+    return this.service$.pipe(
+      switchMap(o => this.getServiceBrokerById(o.entity.service_broker_guid)),
+      combineLatest(this.servicePlanVisibilities$),
+      filter(([p, q]) => !!p && !!q),
+      map(([serviceBroker, allServicePlanVisibilities]) => {
+        if (serviceBroker.entity.space_guid) {
```

Comment:
Does this need a default case to cover if there's no broker space guid or visible service plans, or is this now not possible because we block access in those cases?

Reply:
Method is used by the `Select Plan` form, the user shouldnt be able to get to the form is no visible plans are available. however, since this is in the service I'll add a default case.